lib/connection_controller.go

```diff
@@ -116,6 +117,26 @@ func (cc *ConnectionController) startPersistentConnector() {
 	}
 }
 
+// startValidatorConnector is responsible for ensuring that the node is connected to all active validators. It does
+// this in two steps. First, it looks through the already established connections and checks if any of these connections
+// are validators. If they are, it adds them to the validator index. It also checks if any of the existing validators
+// are no longer active and removes them from the validator index. Second, it checks if any of the active validators
+// are missing from the validator index. If they are, it attempts to connect to them.
+func (cc *ConnectionController) startValidatorConnector() {
+	cc.startGroup.Done()
+	for {
+		select {
+		case <-cc.exitChan:
```

Comment:
AFAIK, trying to read from the exitChan channel here shouldn't work. You already call `case <-cc.exitChan:` in the `startPersistentConnector` event loop above. Only one of the event loops will be able to read from the exitChan event....